server/src/main/java/org/elasticsearch/cluster/coordination/FutureExecutor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.coordination;
+
+import org.elasticsearch.common.unit.TimeValue;
+
+/**
+ * Device which supports running a task after some delay has elapsed.
+ */
+public interface FutureExecutor {
+    /**
+     * Schedule the given task for execution after the given delay has elapsed.
+     */
+    void schedule(TimeValue delay, Runnable task);
```

Comment:
I wonder if we should have this follow the shape of the schedule method in `java.util.concurrent.ScheduledExecutorService`, i.e., have the command before the delay:

```
public ScheduledFuture<?> schedule(Runnable command,
                                       long delay, TimeUnit unit);
```